### PR TITLE
Notifications

### LINKED ISSUE #3: Notification System
**Is your feature request related to a problem? Please describe.**
I'd like to be notified whenever something _happens_ on Träwelling that I might be concerned about. Thinking of:

* Someone followed me
* Someone liked my status
* Someone checked into my connection (previously done with Twitter DMs)
* My Social Post was unsuccessful (Mastodon instance offline, Twitter key revoked, big whale time)

**Describe the solution you'd like**
* "bell button" in Träwelling which shows all read and unread notifications of all types, comparable with Facebook and Twitter.
* Service Worker / PWA-Notifications?

**Describe alternatives you've considered**
Mail notifications? Telegram messages?

**Additional context**

* [Tutorial for Laravels Notification Library](https://www.youtube.com/watch?v=upKOwoe8LsM)
* [Related Gitlab issue](https://gitlab.com/HerrLevin_/trwl/issues/188)

Possible UI of notification center

## This is a larger project that can be implemented in several steps. Please refer this issue whenever you implement one of the given examples.

Please close this issue once all examples are implemented.

### PR DESCRIPTION
**Pull request by jeyemwey**
This PR introduces a notification system for users of the Träwelling website.

Currently supported (and tested) notifications:

_(B receives a notification.)_
* A liked B's check-in.
* A followed B.
* B has checked into a train. A checks into the same train with overlapping stations.
* B has checked into a connection and checked the Tweet or Toot boxes. Something went wrong while posting (e.g. Instance down or `401 Unauthenticed`). - not feature-tested!

Closes #3.

grafik

To Do:
- [x] Notification Center on small phones (< iPhone 6 Plus) is kind of broken.

Stuff for later:
- [ ] Pull/push every n seconds or minutes for new notifications. Currently only watching at pageload.
- [ ] Some kind of idle-animation when marking one or all notifications as read would be great.

### CODE REVIEW

**Comment by jeyemwey** [comment]
And @DC7IA: Could you maybe add 🇸🇪 language files for this PR? I guess that the text strings are not at debate, especially their meaning in other languages.

File: app/Http/Controllers/NotificationController.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public static function toggleReadState($id) {
+        $notification = Auth::user()->notifications->where('id', $id)->first();
+        
+        // Might have cached the html property and would then try to shove it in the DB, mostly
+        // happened during tests.
+        if(isset($notification->html)) unset($notification->html);
+
+        if($notification->read_at == null) { // old state = unread
+            $notification->markAsRead();
+            return response("", 201); // new state = read, 201=created
```
Comment (HerrLevin) [inline app/Http/Controllers/NotificationController.php]: Should return notification

File: app/Http/Controllers/NotificationController.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        } else { // old state = read
+            $notification->markAsUnread();
+            return response("", 202); // new state = unread, 202=accepted
```
Comment (HerrLevin) [inline app/Http/Controllers/NotificationController.php]: should return notification

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -188,6 +189,10 @@ public static function getProfilePage($username) {
 
     }
 
+    /**
```
Comment (HerrLevin) [inline app/Http/Controllers/UserController.php]: a "delete all notifications" is missing from the `DestroyUser`-method

File: app/Http/Controllers/NotificationController.php (hunk 1 above)
Comment (jeyemwey) [reply app/Http/Controllers/NotificationController.php]: Fixed with b9c220e

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (jeyemwey) [reply app/Http/Controllers/UserController.php]: Fixed with b9c220e

File: app/Http/Controllers/NotificationController.php (hunk 2 above)
Comment (jeyemwey) [reply app/Http/Controllers/NotificationController.php]: Fixed with b9c220e

**Pull request merged**